fw/http_parser.c

```diff
@@ -11761,19 +11817,29 @@ tfw_http_parse_resp(void *resp_data, unsigned char *data, unsigned int len,
 			}
 			__FSM_MOVE(Resp_HdrL);
 		case 'p':
-			if (likely(__data_available(p, 19)
-			           && C8_INT_LCM(p + 1, 'r', 'o', 'x', 'y',
-						 '-', 'a', 'u', 't')
-				   && C8_INT_LCM(p + 9, 'h', 'e', 'n', 't',
-						 'i', 'c', 'a', 't')
-				   && TFW_LC(*(p + 17)) == 'e'
-				   && *(p + 18) == ':'))
+			if (likely(__data_available(p, 17)
+			           && C4_INT_LCM(p + 1, 'r', 'o', 'x', 'y')))
 			{
-				__msg_hdr_chunk_fixup(data, __data_off(p + 18));
-				parser->_i_st = &&RGen_HdrOtherV;
-				__msg_hdr_set_hpack_index(48);
-				p += 18;
-				__FSM_MOVE_hdr_fixup(RGen_LWS, 1);
+				if (C8_INT_LCM(p + 5, '-', 'c', 'o', 'n', 'n', 'e', 'c', 't')
+					&& C4_INT3_LCM(p + 13, 'i', 'o', 'n', ':'))
+				{
+					/* Proxy-Connection */
+					__msg_hdr_chunk_fixup(data, __data_off(p + 16));
+					parser->_i_st = &&RGen_HdrOtherV;
+					parser->hdr.flags |= TFW_STR_HBH_HDR;
+					p += 16;
+					__FSM_MOVE_hdr_fixup(RGen_LWS, 1);
+				}
+				if (__data_available(p, 19)
+					&& C8_INT_LCM(p + 5, '-', 'a', 'u', 't', 'h', 'e', 'n', 't')
+					&& C8_INT7_LCM(p + 11, 'n', 't', 'i', 'c', 'a', 't', 'e', ':'))
```

Comment:
Please adjust the too wide lines here and above - we fit 80, maybe 85 if necessary, characters in line